folly/test/StringTest.cpp

```diff
@@ -48,8 +48,8 @@ TEST(StringPrintf, NumericFormats) {
   EXPECT_EQ("5000000000", stringPrintf("%lld", 5000000000LL));
   EXPECT_EQ("-5000000000", stringPrintf("%lld", -5000000000LL));
   EXPECT_EQ("-1", stringPrintf("%d", 0xffffffff));
-  EXPECT_EQ("-1", stringPrintf("%" PRId64, 0xffffffffffffffff));
-  EXPECT_EQ("-1", stringPrintf("%" PRId64, 0xffffffffffffffffUL));
+  EXPECT_EQ("-1", stringPrintf("%" PRId64, 0xffffffffffffffffLL));
+  EXPECT_EQ("-1", stringPrintf("%" PRId64, 0xffffffffffffffffULL));
```

Comment:
Given that the types of these literals are long long and unsigned long long, it seems more correct to use "%lld".

If we also want to explicitly test with 64-bit types we can also add some checks where we explicitly
cast the arguments to 64-bit types.

  EXPECT_EQ("-1", stringPrintf("%" PRId64, static_cast<int64_t>(0xffffffffffffffffLL)));
  EXPECT_EQ("-1", stringPrintf("%" PRId64, static_cast<uint64_t>(0xffffffffffffffffULL)));

Reply:
This looks like it is intended to test the `"%" PRId64` format, so, best to cast the literals.

Reply:
I'm with @yfeldblum here.  On my system, `PRId64` expands to `lld` since I am on a 64-bit machine. I believe the intent of this part of the test is to use the cross-platform specifiers in `inttypes.h`. As such, I'll just cast the literals as you both suggest.